File: src/main/java/com/pisphere/oracles/OracleNetwork.java
package com.pisphere.oracles;

import java.util.HashMap;
import java.util.Map;

public class OracleNetwork {
    private Map<String, Double> priceFeeds;

    public OracleNetwork() {
        priceFeeds = new HashMap<>();
    }

    public void updatePriceFeed(String asset, double price) {
        priceFeeds.put(asset, price);
        System.out.println("Updated price feed for " + asset + ": " + price);
    }

    public double getPriceFeed(String asset) {
        return priceFeeds.getOrDefault(asset, 0.0);
    }
}

File: src/main/python/oracles/oracle_network.py
class OracleNetwork:
    def __init__(self):
        self.price_feeds = {}

    def update_price_feed(self, asset, price):
        self.price_feeds[asset] = price
        print(f"Updated price feed for {asset}: {price}")

    def get_price_feed(self, asset):
        return self.price_feeds.get(asset, 0.0)
